adds sharepoint lists changelog (#5188)
alcionai/corso: CHANGELOG.md

  to the object store when using corso. This affects repos that have many
  backups created in them per day the most.
- Group mailbox emails can now be exported as `.eml` files.

> - Feature Preview: Corso now supports sSharePoint lists with operations including backup, list, details, restore, and export. Utilize `corso backup create sharepoint --site <site-url> --data lists` for Sharepoint list backups using the `--data` flag.

Reviewer: ```suggestion
- Feature Preview: Corso now supports backup, export & restore of SharePoint lists. Lists backup can be initiated using `corso backup create sharepoint --site <site-url> --data lists`
```

- Exchange in-place restore may restore items in well-known folders to different folders if the user has well-known folder names change based on locale and has updated the locale since the backup was created.
- In-place Exchange contacts restore will merge items in folders named "Contacts" or "contacts" into the default folder.
- External users with access through shared links will not receive these links as they are not sent via email during restore.

> - Due to limitations in Graph APIs pertaining to column definition support for certain column types such as `location`, `hyperlink/picture`, and `metadata`, the restoration process for SharePoint list items containing these columns will result in differences compared to the original items.

Reviewer: ```suggestion
- Graph API has limited support for certain column types such as `location`, `hyperlink/picture`, and `metadata`.  Restoring SharePoint list items containing these columns will result in differences compared to the original items.
```

> - Support for SharePoint list item attachments is currently unavailable due to limitations in graph APIss.

Reviewer: ```suggestion
- SharePoint list item attachments are not available due to graph API limitations.
```
